error: runtime error: invalid memory address or nil pointer dereference

#### Version

0.15.0

#### Description

After installing cortex with `$ bash -c "$(curl -sS https://raw.githubusercontent.com/cortexlabs/cortex/0.15/get-cli.sh)"` and running `cortex cluster up`, I get the following error message after providing the aws region I would like to use:

`error: runtime error: invalid memory address or nil pointer dereference` 

I could not figure out why, so I removed this installation and went to the docs and used version 0.14 and the same exact commands seem to work fine. 


#### Configuration

I'll get the message after providing the region if I don't provide a config file but if I do, I'll get the error message right away. 

#### Steps to reproduce

1. `$ bash -c "$(curl -sS https://raw.githubusercontent.com/cortexlabs/cortex/0.15/get-cli.sh`
2. `$ cortex cluster up `
3.  Provide aws region and return 

#### Expected behavior

Move on to ask configuration for s3 bucket, etc. 

#### Actual behavior

#### Screenshots

<img width="514" alt="Screen Shot 2020-04-01 at 3 39 29 PM" src="https://user-images.githubusercontent.com/31977186/78193305-11415280-742f-11ea-8423-2e7558550eaf.png">


#### Stack traces

N/A

My pleasure. 

1.  Here's my output for `aws iam get-user` 
```
{
    "User": {
        "UserId": "493676065566",
        "Arn": "arn:aws:iam::493676065566:root",
        "CreateDate": "2018-06-07T08:09:37Z",
        "PasswordLastUsed": "2020-04-01T17:59:10Z"
    }
}
```

2.   Here's the output I get for the commands you've asked me to run. 
```
(interface {}) <nil>
---------------------
(*iam.GetUserOutput)(({
  User: {
    Arn: "arn:aws:iam::493676065566:root",
    CreateDate: 2018-06-07 08:09:37 +0000 UTC,
    PasswordLastUsed: 2020-04-01 17:59:10 +0000 UTC,
    UserId: "493676065566"
  }
}) {
  _: (struct {}) {
  },
  User: (*iam.User)(({
  Arn: "arn:aws:iam::493676065566:root",
  CreateDate: 2018-06-07 08:09:37 +0000 UTC,
  PasswordLastUsed: 2020-04-01 17:59:10 +0000 UTC,
  UserId: "493676065566"
}) {
    _: (struct {}) {
    },
    Arn: (*string)((len=30) "arn:aws:iam::493676065566:root"),
    CreateDate: (*time.Time)((2018-06-07 08:09:37 +0000 UTC) {
      wall: (uint64) 0,
      ext: (int64) 63663955777,
      loc: (*time.Location)(<nil>)
    }),
    PasswordLastUsed: (*time.Time)((2020-04-01 17:59:10 +0000 UTC) {
      wall: (uint64) 0,
      ext: (int64) 63721360750,
      loc: (*time.Location)(<nil>)
    }),
    Path: (*string)(<nil>),
    PermissionsBoundary: (*iam.AttachedPermissionsBoundary)(<nil>),
    Tags: ([]*iam.Tag) <nil>,
    UserId: (*string)((len=12) "493676065566"),
    UserName: (*string)(<nil>)
  })
})
=====================
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1b5bf0b]

goroutine 1 [running]:
github.com/cortexlabs/cortex/pkg/lib/exit.Panic(0x2e88a80, 0xc00007f680, 0x0, 0x0, 0x0)
	/home/ubuntu/src/github.com/cortexlabs/cortex/pkg/lib/exit/exit.go:60 +0x1c0
github.com/cortexlabs/cortex/pkg/lib/exit.RecoverAndExit(0x0, 0x0, 0x0)
	/home/ubuntu/src/github.com/cortexlabs/cortex/pkg/lib/exit/exit.go:66 +0x9d
panic(0x27f3be0, 0x3e06950)
	/usr/local/go/src/runtime/panic.go:967 +0x166
github.com/cortexlabs/cortex/pkg/lib/aws.(*Client).IsAdmin(0xc00030a070, 0x15)
	/home/ubuntu/src/github.com/cortexlabs/cortex/pkg/lib/aws/iam.go:104 +0x7b
github.com/cortexlabs/cortex/cli/cmd.promptIfNotAdmin(0xc00030a070)
	/home/ubuntu/src/github.com/cortexlabs/cortex/cli/cmd/lib_aws_creds.go:60 +0x59
github.com/cortexlabs/cortex/cli/cmd.getInstallClusterConfig(0xc0000b6f80, 0x14, 0xc0000ac420, 0x28, 0xc0000b6f80, 0x14, 0xc0000ac420, 0x28, 0xc0000ac420, 0x28, ...)
	/home/ubuntu/src/github.com/cortexlabs/cortex/cli/cmd/lib_cluster_config.go:143 +0x10c
github.com/cortexlabs/cortex/cli/cmd.glob..func1(0x3e18540, 0x3e55938, 0x0, 0x0)
	/home/ubuntu/src/github.com/cortexlabs/cortex/cli/cmd/cluster.go:89 +0x10c
github.com/spf13/cobra.(*Command).execute(0x3e18540, 0x3e55938, 0x0, 0x0, 0x3e18540, 0x3e55938)
	/home/ubuntu/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x3e1a0c0, 0xc00065ff40, 0x1, 0x1)
	/home/ubuntu/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/home/ubuntu/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/cortexlabs/cortex/cli/cmd.Execute()
	/home/ubuntu/src/github.com/cortexlabs/cortex/cli/cmd/root.go:132 +0x300
main.main()
	/home/ubuntu/src/github.com/cortexlabs/cortex/cli/main.go:24 +0x20
```

Hope this helps!

--- a/pkg/lib/aws/iam.go
+++ b/pkg/lib/aws/iam.go
@@ -17,6 +17,8 @@ limitations under the License.
 package aws
 
 import (
+	"strings"
+
 	"github.com/aws/aws-sdk-go/service/iam"
 	"github.com/cortexlabs/cortex/pkg/lib/errors"
 )
@@ -50,6 +52,7 @@ func (c *Client) GetGroupsForUser(userName string) ([]iam.Group, error) {
 	return groups, nil
 }
 
+// Note: root users don't have attached policies, but do have full access
 func (c *Client) GetManagedPoliciesForUser(userName string) ([]iam.AttachedPolicy, error) {
 	var policies []iam.AttachedPolicy
 
@@ -87,6 +90,16 @@ func (c *Client) IsAdmin() bool {
 	user, err := c.GetUser()
 	if err != nil {
 		return false
+	}
+
+	// Root users may not have a user name
+	if user.UserName == nil {
+		return true
+	}
+
+	// Root users may have a user name
+	if user.Arn == nil || strings.HasSuffix(*user.Arn, ":root") {
+		return true
 	}
 
 	policies, err := c.GetManagedPoliciesForUser(*user.UserName)

